app/knights/apply_potion.py

```diff
@@ -0,0 +1,13 @@
+class Potion:
+
+    @staticmethod
+    def app_potion(potion: dict, power, hp, protection):
+        if "power" in potion["effect"]:
+            power += potion["effect"]["power"]
+
+        if "protection" in potion["effect"]:
+            protection += potion["effect"]["protection"]
+
+        if "hp" in potion["effect"]:
+            hp += potion["effect"]["hp"]
```

Comment:
These code duplications can be fixed, using `stats = ("protection", "power", "hp")` and looping on `stats`